examples/keras_io/vision/image_classification_using_global_context_vision_transformer.py

```diff
@@ -0,0 +1,1182 @@
+        self.activation = activation
+        self.dropout = dropout
+
+    def build(self, input_shape):
+        self.in_features = input_shape[-1]
+        self.hidden_features = self.hidden_features or self.in_features
+        self.out_features = self.out_features or self.in_features
+        self.fc1 = keras.layers.Dense(self.hidden_features, name="fc1")
+        self.act = keras.layers.Activation(self.activation, name="act")
+        self.fc2 = keras.layers.Dense(self.out_features, name="fc2")
+        self.drop1 = keras.layers.Dropout(self.dropout, name="drop1")
+        self.drop2 = keras.layers.Dropout(self.dropout, name="drop2")
+
+    def call(self, inputs, **kwargs):
+        x = self.fc1(inputs)
+        x = self.act(x)
+        x = self.drop1(x)
+        x = self.fc2(x)
+        x = self.drop2(x)
+        return x
+
+
+"""
+### Stem
+
+> **Notes**: In the code, this module is referred to as **PatchEmbed** but on paper, it
+is referred to as **Stem**.
+
+In the model, we have first used `patch_embed` module. Let's try to understand this
+module. As we can see from the `call` method,
+1. This module first **pads** input
+2. Then uses **convolutions** to extract patches with embeddings.
+3. Finally, uses `ReduceSize` module to first extract features with **convolution** but
+neither reduces spatial dimension nor increases spatial dimension.
+4. One important point to notice, unlike **ViT** or **SwinTransformer**, **GCViT**
+creates **overlapping patches**. We can notice that from the code,
+`Conv2D(self.embed_dim, kernel_size=3, strides=2, name='proj')`. If we wanted
+**non-overlapping** patches then we would've used the same `kernel_size` and `stride`.
+5. This module reduces the spatial dimension of input by $4x$.
+> **Summary:** image $\rightarrow$ padding $\rightarrow$ convolution $\rightarrow$
+(feature_extract + downsample)
+"""
+
+
+class PatchEmbed(keras.layers.Layer):
+    """Patch embedding block.
+
+    Args:
+        embed_dim: feature size dimension.
+    """
+
+    def __init__(self, embed_dim, **kwargs):
+        super().__init__(**kwargs)
+        self.embed_dim = embed_dim
+
+    def build(self, input_shape):
+        self.pad = keras.layers.ZeroPadding2D(1, name="pad")
+        self.proj = keras.layers.Conv2D(self.embed_dim, 3, 2, name="proj")
+        self.conv_down = ReduceSize(keepdims=True, name="conv_down")
+
+    def call(self, inputs, **kwargs):
+        x = self.pad(inputs)
+        x = self.proj(x)
+        x = self.conv_down(x)
+        return x
+
+
+"""
+### Global Token Gen.
+
+> **Notes:** It is one of the two **CNN** modules that is used to imppose inductive bias.
+
+As we can see from above cell, in the `level` we have first used `to_q_global/Global
+Token Gen./FeatExtract`. Let's try to understand how it works,
+
+* This module is series of `FeatureExtract` module, according to paper we need to repeat
+this module $K$ times, where
+$$K = \log_{2}(\frac{H}{h})$$
```

Comment:
Don't use LaTeX syntax. You can use a code block with pseudocode or Python code.